File: 은지/LV2/올바른 괄호.js
function solution(s) {
  var stack = []
  for (let x of s) {
    if (stack.length === 0) {
      if (x === ")") {
        return false
      } else {
        stack.push(x)
      }
    }
    else {
      if (x === "(") {
        stack.push(x)
      } else {
        stack.pop()
      }
    }
  }
  if (stack.length) {
    return false
  }
  return true
}

File: 은지/LV2/올바른 괄호.py
def solution(s):
    stack = []
    for x in s:
        if len(stack) == 0:
            if x == ")":
                return False
            else:
                stack.append(x)
        else:
            if x == "(":
                stack.append(x)
            else:
                stack.pop(-1)
    if len(stack):
        return False
    return True

s = "(()("
print(solution(s))